common/formatting/align.h

```diff
@@ -66,32 +80,56 @@ struct ColumnPositionEntry {
 // and call ReserveNewColumn() in locations that want a new columns.
 class ColumnSchemaScanner : public TreeContextPathVisitor {
  public:
-  ColumnSchemaScanner() = default;
+  ColumnSchemaScanner()
+      : sparse_columns_(ColumnPositionTree({{}, TokenInfo::EOFToken(), {}})) {}
 
   // Returns the collection of column position entries.
-  const std::vector<ColumnPositionEntry>& SparseColumns() const {
-    return sparse_columns_;
-  }
+  const ColumnPositionTree& SparseColumns() const { return sparse_columns_; }
 
  protected:
+  // Returns subpath relative to base_path
+  static SyntaxTreePath GetSubpath(
+      const SyntaxTreePath& base_path,
+      std::initializer_list<SyntaxTreePath::value_type> subpositions) {
+    auto subpath = base_path;
+    subpath.insert(subpath.end(), subpositions);
+    return subpath;
+  }
+
   // Mark the start of a new column for alignment.
+  // 'parent_column' is a reference to the parent column.
   // 'symbol' is a reference to the original source syntax subtree.
   // 'properties' contains alignment configuration for the column.
   // 'path' represents relative position within the enclosing syntax subtree,
   // and is used as a key for ordering and matching columns.
-  void ReserveNewColumn(const Symbol& symbol,
-                        const AlignmentColumnProperties& properties,
-                        const SyntaxTreePath& path);
+  ColumnPositionTree* ReserveNewColumn(
+      ColumnPositionTree& parent_column, const Symbol& symbol,
```

Comment:
As usual, this code-base preferres pointers if the parameter is non-const to indicate that things can be modified.

Reply:
Fixed in [parent PR](https://github.com/chipsalliance/verible/pull/865)